packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -217,19 +217,30 @@ class AttachCommand extends FlutterCommand {
         flutterDevice.startEchoingDeviceLog();
       }
 
+      int result;
       if (daemon != null) {
         AppInstance app;
         try {
-          app = await daemon.appDomain.launch(hotRunner, hotRunner.attach,
-              device, null, true, fs.currentDirectory);
+          app = await daemon.appDomain.launch(
+            hotRunner,
+            hotRunner.attach,
+            device,
+            null,
+            true,
+            fs.currentDirectory,
+          );
         } catch (error) {
           throwToolExit(error.toString());
         }
-        final int result = await app.runner.waitForAppToFinish();
-        if (result != 0)
-          throwToolExit(null, exitCode: result);
+        result = await app.runner.waitForAppToFinish();
+        assert(result != null);
       } else {
-        await hotRunner.attach();
+        result = await hotRunner.attach();
+        assert(result != null);
+      }
+      if (result != 0) {
+        assert(result != null);
+        throwToolExit(null, exitCode: result);
```

Comment:
Is there no meaningful error message we could display here?

Reply:
It's not clear to me what this code is doing. I just refactored it to remove the timeouts. I'll leave this as-is for now.